src/SourceGenerators/Uno.UI.SourceGenerators.Tests.Shared/Given_XBindRewriter.cs

```diff
@@ -1,57 +0,0 @@
-﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
-using System;
-using System.Collections.Generic;
-using System.Linq;
-using Uno.UI.SourceGenerators.XamlGenerator.Utils;
-
-namespace Uno.UI.SourceGenerators.Tests
-{
-	[TestClass]
-	public class Given_XBindRewriter
-	{
-		[TestMethod]
-
-		// DataTemplates (with context)
-		[DataRow("ctx", "MyProperty.A", "ctx.MyProperty.A")]
-		[DataRow("ctx", "MyProperty", "ctx.MyProperty")]
-		[DataRow("ctx", "MyStaticProperty", "ctx.MyStaticProperty")]
-		[DataRow("ctx", "MyStaticMethod()", "ctx.MyStaticMethod()")]
-		[DataRow("ctx", "MyProperty.A.ToLower()", "ctx.MyProperty.A.ToLower()")]
-		[DataRow("ctx", "global::System.String.Format('{0:X8}', a.Value)", "global::System.String.Format('{0:X8}', ctx.a.Value)")]
-		[DataRow("ctx", "Static.MyFunction(42.0)", "ctx.Static.MyFunction(42.0)")]
-		[DataRow("ctx", "Static.MyFunction(true)", "ctx.Static.MyFunction(true)")]
-		[DataRow("ctx", "Static.MyFunction(MyProperty)", "ctx.Static.MyFunction(ctx.MyProperty)")]
-		[DataRow("ctx", "MyNameSpace.Static2.MyProperty", "ctx.MyNameSpace.Static2.MyProperty")]
-		[DataRow("ctx", "MyNameSpace.Static2.MyEnum.EnumMember", "ctx.MyNameSpace.Static2.MyEnum.EnumMember")]
-		[DataRow("ctx", "MyNameSpace.Static2.MyProperty.ToArray()", "ctx.MyNameSpace.Static2.MyProperty.ToArray()")]
-		[DataRow("ctx", "MyNameSpace.Static2.MyFunction(MyProperty)", "ctx.MyNameSpace.Static2.MyFunction(ctx.MyProperty)")]
-		[DataRow("ctx", "MyFunction(MyProperty)", "ctx.MyFunction(ctx.MyProperty)")]
-		[DataRow("ctx", "", "ctx")]
-
-		// Type Casts
-		[DataRow("ctx", "(FontFamily)Value", "(FontFamily)ctx.Value")]
-		[DataRow("ctx", "(FontFamily)a.Value", "(FontFamily)ctx.a.Value")]
-		[DataRow("ctx", "(global::System.Int32)a.Value", "(global::System.Int32)ctx.a.Value")]
-
-		// Pathless casting https://learn.microsoft.com/en-us/windows/uwp/xaml-platform/x-bind-markup-extension#pathless-casting
-		[DataRow("ctx", "MyFunction((global::System.String))", "ctx.MyFunction(((global::System.String)ctx))")]
-		[DataRow("ctx", "MyFunction2((global::System.String),(global::System.String))", "ctx.MyFunction2(((global::System.String)ctx),((global::System.String)ctx))")]
-		[DataRow("ctx", "(global::System.String)", "((global::System.String)ctx)")]
-
-		// Attached properties
-		[DataRow("ctx", "AdornerCanvas.(MyNamespace.FrameworkElementExtensions.ActualWidth)", "MyNamespace.FrameworkElementExtensions.GetActualWidth(ctx.AdornerCanvas)")]
-		[DataRow("ctx", "AdornerCanvas.(Grid.Row)", "Grid.GetRow(ctx.AdornerCanvas)")]
-		[DataRow("ctx", "global::System.String.Format('{0:X8}', AdornerCanvas.(MyNamespace.FrameworkElementExtensions.ActualWidth))", "global::System.String.Format('{0:X8}', MyNamespace.FrameworkElementExtensions.GetActualWidth(ctx.AdornerCanvas))")]
```

Comment:
Will cover

Reply:
I'm not seeing a new test for nested functions, can you point me to one?

Reply:
@Youssef1313 (in case it got lost in all the updates)